### ISSUE TITLE
Non-conformance in names

### ISSUE THREAD

**Issue by jkrasting** [jetesdal/xwmt#28]
@jetesdal and @gmacgilchrist -  there are a lot of variables, attributes, methods, etc. that could have more descriptive names.  As a general rule, they should be longer than 3 characters and be all lower case.

We can exempt a small number if there is a good reason, (e.g. `ds` is commonly used throughout the community to denote an `xarray.core.Dataset` object, but most of these could be made more intuitive.

Can you suggest some alternatives? 

**Variables** -- Must be longer than three characters and all lower case

- ct
- dd
- dm
- dt
- dy
- F
- F_mean
- F_transformed
- G
- _G
- Jlmass
- l
- p
- sa

**Attributes** -- Must be longer than three characters and all lower case

- Cp
- da
- dd
- dl
- ds
- F
- h
- Jl
- l
- Ldot
- lm
- Qm
- tf
- ti
- z

**Functions** -- Must be longer than three characters and all lower case

- calc_F_transformed
- calc_G
- expand_surface_to_3D
- hldot_from_Jl
- hldot_from_Ldot_hldotmass
- Jlmass_from_Qm_lm_l

**Class Methods** -- Must be longer than three characters and all lower case

- calc_Fl
- calc_F_transformed
- calc_G
- dd
- F
- G

**Comment by gmacgilchrist** [comment]
I wonder if retaining some of the capitalization might be important, to be consistent with mathematical nomenclature? For example, Lambda (or L) is the vertically-intergrated version of lambda (l); that is, capitalization denotes the exensive quantitity. Capital J is also a common mathematical notation for tracer fluxes, while capital Q is common for surface mass fluxes. These latter two are more easily replaced.

Anyway, here are suggestions for some of these...

Variables -- Must be longer than three characters and all lower case

- ct : conservative_temp
- dd : datadict
- dm
- dt
- dy
- F : @jetesdal Is this a functionally different quantity to hldot? I remain unsure about this.
- F_mean
- F_transformed
- G : wmt
- _G
- Jlmass : jlambda_mass
- l : lambda
- p : pressure
- sa : absolute_salinity

Attributes -- Must be longer than three characters and all lower case

- Cp : specific_heat_capacity
- da : @jkrasting Keep as da as another accepted terms for DataArray?
- dd : datadict
- dl : delta_lambda
- ds
- F
- h : layer_thickness
- Jl : jlambda
- l : lambda
- Ldot : @jetesdal @jkrasting Tricky one because the capital L is used to denote the extensive quantity (vertically inteegrated lambda). Capitalization is an accepted nomenclature, not sure what else to suggest.
- lm : lambda_in_mass @jetesdal can you confirm that's what this is?
- Qm : mass_flux
- tf
- ti
- z

Functions -- Must be longer than three characters and all lower case

- calc_F_transformed
- calc_G
- expand_surface_to_3D
- hldot_from_Jl
- hldot_from_Ldot_hldotmass
- Jlmass_from_Qm_lm_l

Class Methods -- Must be longer than three characters and all lower case

- calc_Fl
- calc_F_transformed
- calc_G
- dd
- F
- G

**Comment by hdrake** [comment]
> I wonder if retaining some of the capitalization might be important, to be consistent with mathematical nomenclature? For example, Lambda (or L) is the vertically-intergrated version of lambda (l); that is, capitalization denotes the exensive quantitity.

I agree with Graeme that it is more important for the capitalization convention for these terms to be consistent with the mathematical convention that the data formatting conventions, at least internally in the package. 

I can work on implementing @gmacgilchrist's suggested renaming if others agree.

**Comment by hdrake** [comment]
Thanks for the input, @jkrasting. I hadn't known the convention was so strict for variables/functions.
> The recommended way around this is to add descriptive suffixes, e.g. `lambda_vint`
I am satisfied with just using suffixes like this.

**Comment by gmacgilchrist** [comment]
Staying close to conventional standards feels like the right move.

**Comment by hdrake** [comment]
> As a general rule, they should be longer than 3 characters and be all lower case.

Where does this convention come from? I don't see it in the [PEP8](https://peps.python.org/pep-0008/#naming-conventions) style guide.

**Comment by jkrasting** [comment]
You’re right, @hdrake.  PEP-8 does not specify a minimum length, but some Python linters enforce this convention for clarity.  Pylint is good example of this.
